aten/src/ATen/native/Convolution.cpp

```diff
@@ -840,8 +842,8 @@ at::Tensor _convolution(
 
   at::MemoryFormat backend_memory_format = at::MemoryFormat::Contiguous;
 
-  if (detail::getCUDAHooks().compiledWithCuDNN() && cudnn_conv_use_channels_last(input, weight)) {
-    backend_memory_format = (k == 5) ? at::MemoryFormat::ChannelsLast3d : at::MemoryFormat::ChannelsLast;
+  if (detail::getCUDAHooks().compiledWithCuDNN()) {
+    backend_memory_format = cudnn_conv_suggest_memory_format(input, weight);
```

Comment:
Does the `k==5` condition not matter anymore? (e.g., is it covered by the ` can_use_cudnn_channels_last_3d` condition?)

Reply:
Ideally `k==5` should be a subset of `can_use_cudnn_channels_last_3d`. We can add another check to make it safer though

```diff
- if (can_use_cudnn_channels_last_3d)
+ if (can_use_cudnn_channels_last_3d && input.dim() == 5)
```

in the new `cudnn_conv_suggest_memory_format` function